archive/zip_archiver.go

```diff
@@ -83,12 +98,25 @@ func (a *ZipArchiver) ArchiveDir(indirname string) error {
 	defer a.close()
 
 	return filepath.Walk(indirname, func(path string, info os.FileInfo, err error) error {
+		tmpFilePath, _ := filepath.Abs(path)
+		tmpLongFileName, _ := filepath.Abs(indirname)
+		shortFileName := strings.TrimPrefix(tmpFilePath, tmpLongFileName)
+		isMatch := checkMatch(shortFileName, excludes)
 		if info.IsDir() {
+			if isMatch {
```

Comment:
I think you probably still want to walk all the dirs and match against the full relative paths? You may want to add a test case that covers this.

Reply:
If you match on a directory you are excluding the directory why match after that?

i.e.
/hi/
/hello/world
/hello/i/am/here
if you exclude hello/ you should exclude everything in the hello path why still walk?